File: elemi_programozasi_tetelek_rekurzivan/csoportositas/csoportositas.js
function csoportositas(lista){
    let szotar = {};
    for (const elem of lista) {
        let kulcs = elem[0];
        if (szotar.hasOwnProperty(kulcs)) {
            szotar[kulcs].push(elem);
        } else {
            szotar[kulcs] = [elem];
        }
    }
    return szotar
}

let szavak = [ 'répa', 'retek', 'mogyoró', 'korán', 'reggel', 'ritkán', 'rikkant', 'a', 'rigó'];

let szotar = csoportositas(szavak);
for (const kulcs of Object.getOwnPropertyNames(szotar)) {
    console.log(`${kulcs}: ${szotar[kulcs]}`);
}

File: elemi_programozasi_tetelek_rekurzivan/csoportositas/csoportositas.py
def csoportositas(lista):
    szotar = {}
    for elem in lista:
        kulcs = elem[0]
        if kulcs in szotar.keys():
            szotar[kulcs].append(elem)
        else:
            szotar[kulcs] = [elem]
    return szotar

szavak = ['répa', 'retek', 'mogyoró', 'korán', 'reggel', 'ritkán', 'rikkant', 'a', 'rigó']
szotar = csoportositas(szavak)
for kulcs in szotar.keys():
    print(f'{kulcs}: {szotar[kulcs]}')